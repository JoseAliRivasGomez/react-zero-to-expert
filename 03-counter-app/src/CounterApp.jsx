import {useState} from 'react';
import PropTypes from 'prop-types';

const sumar3 = (e, a, x) => {
    a=a+x;
    console.log(e);
    console.log(a);
}

export const CounterApp = ({value}) => {

    const [valor, setValor] = useState(value);

    const sumar = (e) => {
        setValor(valor+1);
        console.log(e);
        console.log(valor);
    }

    const restar = (e) => {
        setValor(valor-1);
        console.log(e);
        console.log(valor);
    }

    const reset = (e) => {
        setValor(value);
        console.log(e);
        console.log(valor);
    }



    const sumar2 = (e, x) => {
        setValor((v) => v+x);
        console.log(e);
        console.log(valor);
    }

    return (
    <>
        <h1>CounterApp</h1>
        <h2>{valor}</h2>
        <button onClick={sumar}>+1</button>
        <button onClick={restar}>-1</button>
        <button aria-label="btn-reset" onClick={reset}>Reset</button>
        <button onClick={(e) => sumar2(e, 2)}>+2</button>
        <button onClick={(e) => sumar3(e, 1, 2)}>1+2</button>
    </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}