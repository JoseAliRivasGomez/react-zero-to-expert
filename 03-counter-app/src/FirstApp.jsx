import PropTypes from 'prop-types';

const message = "Hello World";
const number = 173;
const array = [1,2,3,4,5,6,7,8,9];
const object = {
    name: "Kim",
    email: "kim@gmail.com"
};

const getResult = (a, b) => {
    return a+b;
}

export const FirstApp = ({title="BCS", subtitle, year, month}) => {

    return (
    <>
        <h2>{message}</h2>
        <h2>{number}</h2>
        <h2>{array}</h2>
        <h2>{array}</h2>
        <h2>{object.name}</h2>
        <code>{JSON.stringify(object)}</code>

        <h2>{getResult(2, 4)}</h2>

        <p>Better Call Saul</p>

        <h1 data-testid="test-title">{title}</h1>
        <h2>{subtitle}</h2>
        <h2>{year}</h2>
        <h2>{month}</h2>
    </>
    );
}

FirstApp.propTypes = {
    subtitle: PropTypes.string.isRequired,
    year: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
    month: 99
}