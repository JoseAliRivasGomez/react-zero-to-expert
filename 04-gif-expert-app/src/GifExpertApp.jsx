import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Better Call Kim', 'Better Call Saul', 'Breaking Bad']);

    const addCategory = (newCategory) => {

        if (categories.includes(newCategory)) return;

        //setCategories(categories => [newCategory, ...categories])
        setCategories([newCategory, ...categories])
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory onNewCategory={addCategory} />

            {
                categories.map( (category) => (
                    <GifGrid 
                    key={category} 
                    category={category} />
                ))
            }

        </>
    )
}
