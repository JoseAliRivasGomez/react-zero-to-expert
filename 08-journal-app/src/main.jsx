import React from 'react';
import ReactDOM from 'react-dom/client';
import { JournalApp } from './JournalApp';
import './styles.css';
import { BrowserRouter } from "react-router-dom";
import { store } from './store/store'
import { Provider } from 'react-redux'

ReactDOM.createRoot(document.getElementById('root')).render(
  //<React.StrictMode>
  <Provider store={store}>
    <BrowserRouter>
      <JournalApp />
    </BrowserRouter>
  </Provider>
  //</React.StrictMode>
)
