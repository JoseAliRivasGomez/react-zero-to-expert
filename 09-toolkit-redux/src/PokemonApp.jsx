import { useEffect } from "react"
import { useSelector, useDispatch } from 'react-redux'
import { getPokemons } from "./store/slices/pokemon";

export const PokemonApp = () => {

    const {page, pokemons, isLoading} = useSelector(state => state.pokemons);
    const dispatch = useDispatch();

    useEffect(() => {
      dispatch(getPokemons());
    

    }, [])
    

    return (
        <>
            <h1>PokemonApp page {page}</h1>
            <hr />
            
            { isLoading ? (<span>Loading...</span>) : (
                <ul>
                    {pokemons.map(({name}) => (
                        <li key={name}>{name}</li>
                    ))}
                </ul>
            )}

            

            <button disabled={isLoading} onClick={() => dispatch(getPokemons(page))}>
                Next
            </button>
        </>
    )
}
