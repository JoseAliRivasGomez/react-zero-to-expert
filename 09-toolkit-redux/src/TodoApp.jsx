import { useState } from "react";
import { useGetTodoQuery, useGetTodosQuery } from "./store/apis/todosApi"

export const TodoApp = () => {

    //const {data: todos, isLoading} = useGetTodosQuery();
    
    const [todoId, setTodoId] = useState(1)
    const {data: todo, isLoading} = useGetTodoQuery(todoId);

    const prevTodo = () => {
        if(todoId === 1) return;
        setTodoId(todoId-1);
    }
    const nextTodo = () => {
        setTodoId(todoId+1);
    }

    return (
        <>
            <h1>TODOs - RTK Query</h1>
            <hr />
            <h4>isLoading: {isLoading ? 'True' : 'False'}</h4>

            <pre>{JSON.stringify(todo)}</pre>



            <button onClick={prevTodo}>
                Prev TODO
            </button>
            <button onClick={nextTodo}>
                Next TODO
            </button>

            {/* <ul>
                {
                    todos.map(({id, title, completed}) => (
                        <li key={id}><strong>{completed ? 'DONE:' : 'Pending:'}</strong> {title}</li>
                    ))
                }
            </ul> */}

            
        </>
    )
}
