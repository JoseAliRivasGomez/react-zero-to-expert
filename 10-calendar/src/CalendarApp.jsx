import { AppRouter } from "./router/AppRouter"
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "./store/store";

export const CalendarApp = () => {
  return (
    <>
      <Provider store={store}>
        <BrowserRouter>
          <AppRouter />
        </BrowserRouter>
      </Provider>
    </>
  )
}
