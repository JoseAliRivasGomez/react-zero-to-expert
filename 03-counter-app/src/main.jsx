import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from "./FirstApp";
import { CounterApp } from './CounterApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title="Breaking Bad" subtitle={123}/> */}
        <CounterApp value={123}/>
    </React.StrictMode>
)