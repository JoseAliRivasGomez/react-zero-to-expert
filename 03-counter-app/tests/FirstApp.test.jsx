import { render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"


describe('Pruebas en First App', () => { 

    // test('Debe hacer match con el snapshot', () => { //No hacer snapshot en produccion

    //     const title = 'BCS';
    //     const subtitle = 'BB';
    //     const year = 1999;

    //     const {container} = render(<FirstApp title={title} subtitle={subtitle} year={year} />)

    //     //console.log(container);
    //     expect(container).toMatchSnapshot();

    //  });

     test('Debe mostrar el titulo en un h1', () => { 

        const title = 'BCS';
        const subtitle = 'BB';
        const year = 1999;

        const {container, getByText, getByTestId} = render(<FirstApp title={title} subtitle={subtitle} year={year} />)

        expect(getByText(title)).toBeTruthy(); //que exista el titulo

        expect(getByTestId('test-title').innerHTML).toContain(title); //Que el p contenga ese texto

        // const p = container.querySelector('p');
        // console.log(p.innerHTML);
        // expect(p.innerHTML).toBe('Better Call Saul'); //Que el p tenga ese texto exacto
        // expect(p.innerHTML).toContain('Better Call Saul'); //Que el p contenga ese texto

     });

     test('Debe mostrar el subtitulo enviado por props', () => { 
        
        const title = 'BCS';
        const subtitle = 'BB';
        const year = 1999;

        const {getAllByText} = render(<FirstApp title={title} subtitle={subtitle} year={year} />)

        expect(getAllByText(subtitle).length).toBe(1); //que haya 1 subtitulo

      })

 })