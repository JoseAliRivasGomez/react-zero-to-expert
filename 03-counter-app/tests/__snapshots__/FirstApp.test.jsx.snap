// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en First App Debe hacer match con el snapshot 1`] = `
<div>
  <h2>
    Hello World
  </h2>
  <h2>
    173
  </h2>
  <h2>
    1
    2
    3
    4
    5
    6
    7
    8
    9
  </h2>
  <h2>
    1
    2
    3
    4
    5
    6
    7
    8
    9
  </h2>
  <h2>
    Kim
  </h2>
  <code>
    {"name":"Kim","email":"kim@gmail.com"}
  </code>
  <h2>
    6
  </h2>
  <p>
    Better Call Saul
  </p>
  <h1
    data-testid="test-title"
  >
    BCS
  </h1>
  <h2>
    BB
  </h2>
  <h2>
    1999
  </h2>
  <h2>
    99
  </h2>
</div>
`;
